src/ctap/crypto_wrapper.rs

```diff
@@ -143,22 +152,33 @@ impl PrivateKey {
         Some(Self::Ed25519(ed25519_compact::KeyPair::from_seed(seed).sk))
     }
 
-    /// Returns the corresponding public key.
-    pub fn get_pub_key(&self) -> CoseKey {
+    /// Returns the ECDSA private key.
+    pub fn ecdsa_key(&self, env: &mut impl Env) -> Option<ecdsa::SecKey> {
         match self {
-            PrivateKey::Ecdsa(ecdsa_key) => CoseKey::from(ecdsa_key.genpk()),
+            PrivateKey::Ecdsa(seed) => ecdsa_key(env, seed),
+            #[allow(unreachable_patterns)]
+            _ => None,
+        }
+    }
+
+    /// Returns the corresponding public key.
+    pub fn get_pub_key(&self, env: &mut impl Env) -> Option<CoseKey> {
```

Comment:
I wonder if this should rather be an internal condition, that all private keys are always valid.

Reply:
Ah right, we call `ok()` here. Why not make it `Result` then?

Reply:
But which error should we use?

Reply:
Don't we usually use VENDOR_INTERNAL for that?

Reply:
I guess it depends on the exact problem, but one of the 2 VENDOR_ errors?